biojava-structure/src/main/java/org/biojava/nbio/structure/io/cif/CifStructureConsumerImpl.java

```diff
@@ -858,8 +860,14 @@ public void consumeStruct(Struct struct) {
         }
 
         if (struct.isDefined() && struct.getEntryId().isDefined()) {
-            pdbHeader.setIdCode(struct.getEntryId().get(0));
-            structure.setPDBCode(struct.getEntryId().get(0));
+            PDBId pdbId = null;
+			try {
+				pdbId = new PDBId(struct.getEntryId().get(0));
+			} catch (PDBIdException e) {
+				throw new IllegalArgumentException(e);
```

Comment:
Is IllegalArgumentException the best RuntimeException to wrap here?

Reply:
I don't know. I didn't want to propagate a checked exception in functions which I didn't explore well.
Any suggestions? @sbliven